week-1/L-functions-nested/exercise2.js

```diff
@@ -1,5 +1,20 @@
+function greeting() {
+  var hello = "Hello";
+  return hello.toUpperCase();
+}
+
+function nameUpperCase(name) {
+  return name.toUpperCase();
+}
+
 var mentor1 = "Daniel";
 var mentor2 = "Irina";
 var mentor3 = "Mimi";
 var mentor4 = "Rob";
 var mentor5 = "Yohannes";
+
+console.log(greeting() + " " + nameUpperCase(mentor1));
+console.log(greeting() + " " + nameUpperCase(mentor2));
+console.log(greeting() + " " + nameUpperCase(mentor3));
+console.log(greeting() + " " + nameUpperCase(mentor4));
+console.log(greeting() + " " + nameUpperCase(mentor5));
```

Comment:
The function `greeting` isn't really doing much here. It just creates the constant string `"HELLO"` every time you call it. As the readme says for this exercise, you can call functions from within other functions, which is what I think it wants you to do here. So you could write something like

```
function greeting(name) {
  return "HELLO " + upperCase(name);
}

function upperCase(name) {
  return name.toUpperCase();
}
```
Then you only need to call `greeting(mentor1)` etc.

Of course we don't really need to write functions like `upperCase` since it's just wrapping the `toUpperCase()` method but it's just an example.

```

Reply:
In the example above in the first function another function was used before it was declared. Does the order matter in this case?